src/components/PasswordInput/PasswordInput.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'react-emotion';
+
+import { InputWrapper } from '../InputWrapper';
+import { Input as StandardInput } from '../Input';
+import { SvgButton } from '../SvgButton';
+import State from '../State/State';
+import RevealIconSvg from './eye.svg';
+import HideIconSvg from './eye-off.svg';
+
+const Input = styled(StandardInput, { label: 'password-input__input' })(
+  ({ theme }) => `
+    padding-right: ${theme.spacings.peta};
+  `
+);
+
+const stylesIcon = css`
+  label: password-input__icon;
+  position: absolute;
+  top: 50%;
+  right: 12px;
+  transform: translateY(-50%);
+`;
+
+const RevealIcon = styled(RevealIconSvg, { label: 'reveal-icon' })(stylesIcon);
+
+const HideIcon = styled(HideIconSvg, { label: 'hide-icon' })(stylesIcon);
+
+/**
+ * PasswordInput component for forms.
+ */
+const PasswordInput = ({ disabled, ...props }) => (
+  <State
+    initialState={false}
+    stateName="isVisible"
+    stateUpdaterName="onToggle"
+    stateUpdater={isVisible => !isVisible}
+  >
+    {({ isVisible, onToggle }) => (
+      <InputWrapper {...{ disabled }} data-selector="password-input">
+        <Input
+          {...props}
+          {...{ disabled }}
```

Comment:
What does this second line do?

Reply:
I've sort of picked this syntax for easily putting variables as props on a component. Writing `{...{ myPropVariable }}` feels much nicer than `myPropVariable={myPropVariable}`. This becomes more evident when you had to destructure a bunch of props in the parent and then want to pass them all down again (i.e., `{...{ propOne, propTwo, propThree }}`. I think I also prefer this to creating props objects in the function before writing out JSX, because it doesn't require giving your functional component  a body for nothing but creating these prop objects.

I guess I could have also done `{...{ ...props, disabled }}` in this case. 😉 

What do you think?

Reply:
I found the syntax slightly unintuitive at first, but I like it as it's shorter.